File: src/Client/X11DisplayDevice.c
#include "DisplayDevice.h"
#if CC_BUILD_X11
#include "Typedefs.h"
#include "ErrorHandler.h"
#include <X11/Xlib.h>

void DisplayDevice_Init(void) {
	Display* display = XOpenDisplay(NULL);
	if (display == NULL) {
		ErrorHandler_Fail("Failed to open display");
	}

	int screen = XDefaultScreen(display);
	Window rootWin = XRootWindow(display, screen);

	/* TODO: Use Xinerama and XRandR for querying these */
	int screens = XScreenCount(display), i;
	for (i = 0; i < screens; i++) {
		struct DisplayDevice device = { 0 };
		device.Bounds.Width  = DisplayWidth(display, i);
		device.Bounds.Height = DisplayHeight(display, i);
		device.BitsPerPixel = DefaultDepth(display, i);
		if (i == screen) DisplayDevice_Default = device;
	}

	DisplayDevice_Meta[0] = display;
	DisplayDevice_Meta[1] = screen;
	DisplayDevice_Meta[2] = rootWin;
}
#endif

File: ClassicalSharp/OpenTK/Platform/X11/X11DisplayDevice.cs
#region License
/* Licensed under the MIT/X11 license.
 * Copyright (c) 2006-2008 the OpenTK Team.
 * This notice may not be removed from any source distribution.
 * See license.txt for licensing detailed licensing details.
 */
#endregion

using System;

namespace OpenTK.Platform.X11 {
	
	internal static class X11DisplayDevice {
		
		internal static void Init() {
			IntPtr display = API.DefaultDisplay;
			int screen = API.DefaultScreen;
			DisplayDevice device = new DisplayDevice();
			
			/* TODO: Use Xinerama and XRandR for querying these */			
			device.Bounds.Width  = API.XDisplayWidth(display, screen);
			device.Bounds.Height = API.XDisplayHeight(display, screen);
			device.BitsPerPixel  = API.XDefaultDepth(display, screen);		
			DisplayDevice.Default = device;
		}
	}
}
